python_modules/libraries/dagster-components/dagster_components/lib/dbt_project/component.py

```diff
@@ -10,48 +10,56 @@
     DbtProject,
     dbt_assets,
 )
-from pydantic import BaseModel
-from typing_extensions import Self
 
 from dagster_components import Component, ComponentLoadContext
 from dagster_components.core.component import component_type
+from dagster_components.core.schema.base import ResolvableModel
 from dagster_components.core.schema.metadata import ResolvableFieldInfo
 from dagster_components.core.schema.objects import (
     AssetAttributesModel,
     AssetSpecTransformModel,
-    OpSpecBaseModel,
+    OpSpecModel,
     TemplatedValueResolver,
 )
 from dagster_components.lib.dbt_project.scaffolder import DbtProjectComponentScaffolder
 from dagster_components.utils import ResolvingInfo, get_wrapped_translator_class
 
 
-class DbtProjectParams(BaseModel):
+class DbtProjectParams(ResolvableModel["DbtProjectComponent"]):
     dbt: DbtCliResource
-    op: Optional[OpSpecBaseModel] = None
+    op: Optional[OpSpecModel] = None
     asset_attributes: Annotated[
         Optional[AssetAttributesModel], ResolvableFieldInfo(required_scope={"node"})
     ] = None
     transforms: Optional[Sequence[AssetSpecTransformModel]] = None
 
+    def resolve(self, resolver: TemplatedValueResolver) -> "DbtProjectComponent":
+        return DbtProjectComponent(
+            resource=self.dbt,
+            op_spec=self.op.resolve(resolver) if self.op else None,
+            translator=get_wrapped_translator_class(DagsterDbtTranslator)(
+                resolving_info=ResolvingInfo(
+                    "node", self.asset_attributes or AssetAttributesModel(), resolver
+                )
+            ),
+            transforms=[transform.resolve(resolver) for transform in (self.transforms or [])],
+        )
+
```

Comment:
we are going to want to copy up with some pattern to avoid the need to call `resolve` manually all over the place here